cli/utils.go

```diff
@@ -107,7 +105,7 @@ func parseLabels(inputLabels []string) (models.LabelSet, error) {
 	labelSet := make(models.LabelSet, len(inputLabels))
 
 	for _, l := range inputLabels {
-		matcher, err := labels.ParseMatcher(l)
+		matcher, err := compat.Matcher(l)
```

Comment:
Ok, after further inspection, I think it's worth stopping here for a moment to address these inconsistencies.

I see that:

1. `alert_add.go` uses `parseLabels` which under the hood uses `compat.Matcher`
2. `alert_query.go` uses `compat.Matcher` directly.
3. `silence_add.go` uses `parseMatchers` which under the hood uses `compat.Matcher`
4. `silence_query.go` uses `parseMatchers` which under the hood uses `compat.Matcher`

My opinion is that we should get rid of `parseMatchers` and `parseLabels` in `utils.go` and instead have everything in the CLI use `compat.*` directly. 

With that, we can leave the introduction of `compat.Label` for later but at least we're certain that we don't have this weird split of a direct use of `compat.*` and the util functions in `parse*`.

Reply:
> I also don't love the fact that we execute this check as part of parseLabels:

Yeah! The issue here is `amtool` is misusing the label matchers parser to parse labels, despite labels and label matchers being quite different. I wasn't planning to address that in this work though as the root issue that needs to be addressed is finding a way to properly parse labels, instead of misuing the label matchers parser to parse input that isn't really a label matcher.

> My opinion is that we should get rid of parseMatchers and parseLabels in utils.go and instead have everything in the CLI use compat.* directly.

So would we inline the code (copy it) from `utils.go` in `alert_add.go`, `silence_add.go` and `silence_query.go`?

Reply:
> So would we inline the code (copy it) from utils.go in alert_add.go, silence_add.go and silence_query.go?

I believe so, I think it's more "clean" in a way to execute the refactor later as it is more obvious. At least everything now depends on `compat`.